app/controllers/works_controller.rb

```diff
@@ -4,109 +4,88 @@
-    @datacite = Datacite::Resource.new_from_json(work.data_cite)
+    @work = Work.find(params[:id])
+    @datacite = Datacite::Resource.new_from_json(@work.data_cite)
 
-    if @dataset.doi
-      service = S3QueryService.new(@dataset.doi)
+    if @work.doi
+      service = S3QueryService.new(@work.doi)
       @files = service.data_profile
     end
   end
 
   def edit
-    @dataset = Dataset.find(params[:id])
-    work = Work.find(@dataset.work_id)
-    @datacite = Datacite::Resource.new_from_json(work.data_cite)
+    @work = Work.find(params[:id])
+    @datacite = Datacite::Resource.new_from_json(@work.data_cite)
     if @datacite.main_title.nil?
       @datacite.titles << Datacite::Title.new(title: "Enter title here")
     end
   end
 
   def update
-    @dataset = Dataset.find(params[:id])
+    @work = Work.find(params[:id])
     respond_to do |format|
-      update_work_record
-      # And then update the dataset
-      if @dataset.update(dataset_params)
-        format.html { redirect_to dataset_url(@dataset), notice: "Dataset was successfully updated." }
-        format.json { render :show, status: :ok, location: @dataset }
+      work_params = {
+        title: params[:title_main],
+        collection_id: params[:collection_id],
+        data_cite: datacite_resource_from_form
+      }
+      if @work.update(work_params)
+        format.html { redirect_to work_url(@work), notice: "Work was successfully updated." }
+        format.json { render :show, status: :ok, location: @work }
       else
-        work = Work.find(@dataset.work_id)
-        @datacite = Datacite::Resource.new_from_json(work.data_cite)
+        @datacite = Datacite::Resource.new_from_json(@work.data_cite)
         format.html { render :edit, status: :unprocessable_entity }
-        format.json { render json: @dataset.errors, status: :unprocessable_entity }
+        format.json { render json: @work.errors, status: :unprocessable_entity }
       end
     end
   end
 
   def approve
-    dataset = Dataset.find(params[:id])
-    dataset.work.approve(current_user)
-    redirect_to dataset_path(dataset)
+    work = Work.find(params[:id])
+    work.approve(current_user)
+    redirect_to work_path(work)
   end
 
   def withdraw
-    dataset = Dataset.find(params[:id])
-    dataset.work.withdraw(current_user)
-    redirect_to dataset_path(dataset)
+    work = Work.find(params[:id])
+    work.withdraw(current_user)
+    redirect_to work_path(work)
   end
 
   def resubmit
-    dataset = Dataset.find(params[:id])
-    dataset.work.resubmit(current_user)
-    redirect_to dataset_path(dataset)
+    work = Work.find(params[:id])
+    work.resubmit(current_user)
+    redirect_to work_path(work)
   end
 
-  # Outputs the Datacite XML representation of the dataset
+  # Outputs the Datacite XML representation of the work
   def datacite
-    @dataset = Dataset.find(params[:id])
-    work = Work.find(@dataset.work_id)
+    work = Work.find(params[:id])
```

Comment:
This is example of the logic that is not needed anymore. We don't have to look up the `dataset` and then the `work`. The work is all we need.